GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -61,14 +64,44 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
                 StatusId = AddStatus(scenario, featureFile),
                 WorkflowId = AddWorkflow(scenario, featureFile),
                 Description = AddDescription(scenario, featureFile),
-                Scenario = AddScenario(scenario, featureFile)
+                Scenario = AddScenario(scenario, featureFile),
+                Tags = AddTags(scenario, featureFile)
             },
             StepsAttachments = AddStepAttachments(scenario, featureFile)
         };
 
         return createTestCaseRequestExtended;
     }
 
+    private List<Tag> AddTags(Scenario scenario, IFeatureFile featureFile)
+    {
+        var allTags = GherkinHelper.GetAllTags(scenario, featureFile);
+        //Remove tags that will duplicate existing fields
+        allTags.RemoveAll(tag => tag.Name.Contains(TagsConstants.Reference, StringComparison.InvariantCultureIgnoreCase));
+        allTags.RemoveAll(tag => tag.Name.Contains(TagsConstants.Automated, StringComparison.InvariantCultureIgnoreCase));
+        allTags.RemoveAll(tag => tag.Name.Contains(TagsConstants.Status, StringComparison.InvariantCultureIgnoreCase));
+        
```

Comment:
These three calls are asking to be extracted as a method - they're doing pretty same stuff and separate method with a clear name could explain their purpose better than the comment you've left

Reply:
Extracted